api/v1/policy_webhook.go

```diff
@@ -49,18 +70,43 @@ func (r *Policy) ValidateDelete() (admission.Warnings, error) {
 }
 
 // validate the policy name and namespace length
-func (r *Policy) validateName() (admission.Warnings, error) {
+func (r *Policy) validateName() error {
 	policylog.Info("Validating the policy name through a validating webhook")
 
 	// replicated policies don't need pass this validation
 	if _, ok := r.GetLabels()["policy.open-cluster-management.io/root-policy"]; ok {
-		return nil, nil
+		return nil
 	}
 
 	// 1 character for "."
 	if (utf8.RuneCountInString(r.Name) + utf8.RuneCountInString(r.Namespace)) > 62 {
-		return nil, errName
+		return errName
 	}
 
-	return nil, nil
+	return nil
+}
+
+// validate the remediationAction field of the root policy and its policy templates
+func (r *Policy) validateRemediationAction() error {
+	policylog.Info("Validating the policy and config policy remediation action through a validating webhook")
+
+	if r.Spec.RemediationAction != "" {
+		return nil
+	}
+
+	plcTemplates := r.Spec.PolicyTemplates
+	objUnstruct := &unstructured.Unstructured{}
+
+	for _, obj := range plcTemplates {
+		_ = json.Unmarshal(obj.ObjectDefinition.Raw, objUnstruct)
```

Comment:
Could you define `objUnstruct` in the loop instead? I worry about side effects of using a previously filled in `Unstructured`.